projects/fsastorefrontlib/src/cms-components/change-process/change-coverage/change-coverage.component.ts

```diff
@@ -26,40 +22,37 @@ export class ChangeCoverageComponent extends AbstractChangeProcessStepComponent
               optionalProduct => !optionalProduct.isMandatory
             );
             if (optionalProducts && !this.isSimulated(changeRequestData)) {
-              optionalProducts.map((coverage, index) => {
-                const coverageCopy = { ...coverage, index };
-                this.optionalCoverages.push(coverageCopy);
-                this.initialOptionalCoverages.push({ ...coverageCopy });
-              });
+              this.optionalCoverages = [...optionalProducts];
             }
           })
         )
         .subscribe()
     );
   }
 
-  addCoverage(coverage: any) {
-    this.optionalCoverages.forEach(potentialCoverage => {
-      if (
-        potentialCoverage.coverageProduct.code === coverage.coverageProduct.code
-      ) {
-        potentialCoverage.coverageIsIncluded = true;
-      }
-    });
+  resolveCoverageChange(index: number) {
+    if (!this.checkCoverageChanged.includes(index)) {
+      this.checkCoverageChanged.push(index);
+    } else {
+      this.checkCoverageChanged.splice(
+        this.checkCoverageChanged.findIndex(idx => idx === index),
+        1
+      );
+    }
   }
 
-  removeCoverage(coverage: any) {
-    this.optionalCoverages.forEach(potentialCoverage => {
-      if (
-        potentialCoverage.coverageProduct.code === coverage.coverageProduct.code
-      ) {
-        potentialCoverage.coverageIsIncluded = false;
-      }
-    });
+  toggleCoverage(idx: number, newCoverageStatus: boolean) {
+    this.resolveCoverageChange(idx);
+    const included = { ...this.optionalCoverages[idx] };
+    included.coverageIsIncluded = !newCoverageStatus;
+    this.optionalCoverages = this.optionalCoverages.map(
+      (potentialCoverage, index) =>
+        idx !== index ? potentialCoverage : included
+    );
   }
 
   simulateChanges(changeRequestData) {
-    if (this.isPolicyChanged()) {
+    if (this.checkCoverageChanged.length > 0) {
```

Comment:
Additional check for determining whether user can continue (if button disabled is manipulated).